app/workers/backend_metric_worker.rb

```diff
@@ -1,44 +1,19 @@
 # frozen_string_literal: true
 
-class BackendMetricWorker
-  class LockError < StandardError; end
+class BackendMetricWorker < ApplicationJob
+  include Sidekiq::Throttled::Worker
 
-  include Sidekiq::Worker
-  include ThreeScale::SidekiqLockWorker
-  include ThreeScale::SidekiqRetrySupport::Worker
+  queue_as :backend_sync
 
-  sidekiq_options queue: :backend_sync,
-                  retry: 3,
-                  lock: {
-                    timeout: 1.hour.in_milliseconds,
-                    name: proc { |*args| BackendMetricWorker.lock_workers(*args) }
-                  }
-  sidekiq_retry_in do |_count|
-    5.minutes.to_i
-  end
-
-  def self.lock_workers(service_id, metric_id, *)
-    "service:#{service_id}/metric:#{metric_id}"
-  end
-
-  def perform(backend_id, metric_id, *args)
-    retry_job(backend_id, metric_id, *args) unless lock(backend_id, metric_id).acquire!
-
-    begin
-      save_or_delete_metric(backend_id, metric_id)
-    ensure
-      lock.release!
-    end
-  end
-
-  protected
-
-  def retry_job(backend_id, metric_id, *args)
-    raise LockError if last_attempt?
-    self.class.perform_async(backend_id, metric_id, *args)
-  end
+  sidekiq_throttle({
+                     concurrency: {
+                       limit: 1,
+                       key_suffix: ->(service_id, metric_id, *) { "service:#{service_id}/metric:#{metric_id}" },
+                       ttl: 15.minutes.to_i
```

Comment:
The lock was 1 hour before now it is 15 minutes.

https://github.com/3scale/porta/pull/1498/files#diff-81adfba00f06f2460ffd621178a978b5L13

Reply:
No special reason. I didn't know the ideal number. I tried it and seemed to work properly and that number sounds reasonable to me. But I definitely don't mind any number in particular.